src/frontend/layouts/Main/Main.js

```diff
@@ -11,10 +19,38 @@ const Main = () => {
         <Route path="/signup" element={<SignUpPage />} />
       </Route>
       <Route element={<SidebarWithHeader />}>
-        <Route path="/profile" element={<>Profile</>} />
-        <Route path="/" element={<>Feed</>} />
-        <Route path="/explore" element={<>Explore</>} />
-        <Route path="/bookmarks" element={<>Bookmarks</>} />
+        <Route
+          path="/profile"
+          element={
+            <RequireAuth>
+              <ProfilePage />
+            </RequireAuth>
+          }
+        />
+        <Route
+          path="/"
+          element={
+            <RequireAuth>
+              <FeedPage />
+            </RequireAuth>
+          }
+        />
+        <Route
+          path="/explore"
+          element={
+            <RequireAuth>
```

Comment:
you can make use of the outlet in react-router and wrap all requiredAuth into single route like this 
<Route element ={RequiredAuth}>
    <Route  path="/"element={<FeedPage />} />
</Route>

Reply:
Will surely look into this. Thank you.